packages/components/datepicker/src/Datepicker.test.tsx

```diff
@@ -1,11 +1,131 @@
 import React from 'react';
-import { render } from '@testing-library/react';
+import {
+  fireEvent,
+  render,
+  screen,
+  waitFor,
+  within,
+} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
 import { Datepicker } from './Datepicker';
+import { format } from 'date-fns';
+import { act } from 'react-dom/test-utils';
+import { axe } from 'jest-axe';
 
 describe('Datepicker', function () {
+  const testDate = new Date('2022-04-15');
+
   it('renders', () => {
     const tree = render(<Datepicker onSelect={jest.fn()} />);
 
     expect(tree).toBeTruthy();
   });
+
+  it('renders the component with an additional class name', () => {
+    const additionalClassName = 'my-extra-class';
+    render(<Datepicker onSelect={jest.fn()} className={additionalClassName} />);
+
+    expect(
+      screen
+        .getByTestId('cf-ui-datepicker')
+        .classList.contains(additionalClassName),
+    ).toBeTruthy();
+  });
+
+  it('renders with default date set', () => {
+    render(<Datepicker selected={testDate} onSelect={jest.fn()} />);
+
+    expect(screen.getByTestId('cf-ui-datepicker-input')).toHaveValue(
+      format(testDate, 'dd LLL yyyy'),
+    );
+  });
+
+  it('opens calendar when button is clicked', async () => {
+    render(<Datepicker selected={testDate} onSelect={jest.fn()} />);
+
+    act(() => {
+      userEvent.click(screen.getByRole('button'));
+    });
+
+    await waitFor(() => {
+      expect(screen.getByTestId('cf-ui-datepicker')).toHaveAttribute(
+        'aria-expanded',
+        'true',
+      );
+      expect(screen.getByTestId('cf-ui-popover-content')).toBeInTheDocument();
+      expect(screen.getByText(format(testDate, 'LLLL yyyy'))).toBeTruthy();
+    });
+  });
+
+  it('renders the calendar initialy open and close when esc key is pressed', async () => {
```

Comment:
I think this test should be split into two cases, one about `defaultIsOpen` and the other for closing on escape.